lib/srv/desktop/tdp/conn.go

```diff
@@ -57,3 +76,18 @@ func (c *Conn) OutputMessage(m Message) error {
 	}
 	return nil
 }
+
+// SendError is a convenience function for sending an error message.
+func (c *Conn) SendError(message string) error {
+	return c.OutputMessage(Error{Message: message})
+}
+
+// LocalAddr returns local address
+func (c *Conn) LocalAddr() net.Addr {
```

Comment:
What happens if a Conn without Local and Remote is used in place of a TrackingConn? Should we use separate constructors and types so it can't be misused?

Reply:
I made `tdp.Conn` implement `srv.TrackingConn` with these fields so we could use the `tdp.Conn` directly with our idle connection monitor.

Prior to this change, the `tdp.Conn` wrapped a `tls.Conn`, and we passed the underlying `tls.Conn` to the connection monitor. This makes it hard to do a nice clean shutdown as the monitor would close the `tls.Conn` out from underneath you, before you had a chance to notify the end user why the connection is being closed.

If you don't specify the local/remote fields, the `tdp.Conn` still parses messages and runs a remote desktop session as usual, but places that expect a `TrackingConn` will omit those fields in any audit events that they emit.

We could make them required, but I kind of like that a `tdp.Conn` just wraps an arbitrary `ReadWriter` - nice for testing without making a network connection.

Reply:
So sounds like a TrackingConn is a nice-to-have, but it won't break anything if a non-tracking one is used. Fair enough.